### DISCUSSION TITLE
Ändern der Konfigurationsdateien

### DISCUSSION THREAD

**Discussion by Hotzi57** [flobz/psa_car_controller#739] [Q&A] [answered]
![Screenshot 2024-01-25 114436](https://github.com/flobz/psa_car_controller/assets/157227065/040180b4-40d3-4879-949e-5f4763359388)
Mir ist es nicht möglich, die Konfigurationsdateien zu ändern.  Ich bin ein absoluter Homeassistant Neuling uind ich habe mir diesen nur installiert, um den PSA-Car-Controller zu nutzen.
Home Assistant:
Core:2024.1.5
Supervisor 2023.12.1
OS 11.4
Frontend 20240104.0
Hardware Raspberry PI5
Das System läuft fehlerfrei, PSA-Car-Controller ebenfalls. Ich habe Avanced SSH&Web Terminal installiert und damit auch den "docker..." Befehl aus der README.md von flobz ausgeführt. Ergebniss: "/mnt/data/supervisor/addons/data/b9f12d83_psacc"
Hier komme ich nicht weiter, weil mit cd <Path> nichts geht. mnt kann ich noch erreichen, ber dann ist es vorbei /data gibt es schon nicht mehr. 
Ich habe noch keine Gelegenheit gehabt, das HA Betriebssystem zu erlernen, aber mit Debian bin ich etwas vertraut.
Kann mir bitte jemand helfen, damit ich meinen Autotyp einpflegen kann?
Vielen Dank im Vorraus.
Nochmal in Google-Englisch:
I am unable to change the configuration files. I am an absolute home assistant newbie and I only installed it to use the PSA car controller.
Home Assistant:
Core:2024.1.5
Supervisor 2023.12.1
OS 11.4
Frontend 20240104.0
Hardware Raspberry PI5
The system runs error-free, as does the PSA car controller. I installed Advanced SSH&Web Terminal and used it to run the "docker..." command from flobz's README.md. Result: "/mnt/data/supervisor/addons/data/b9f12d83_psacc"
I can't get any further here because nothing works with CD. I can still reach mnt, but then it's over / data no longer exists.
I haven't had a chance to learn the HA operating system yet, but I am somewhat familiar with Debian.
Can someone please help me so I can enter my car type?
Many thanks in advance.

**Comment by mitasa** [comment, marked as answer]
The PSA car controller is installed in a docker container. You can't cd to it directly, but instead use docker, first find out the contaner ID by typing

docker ps

this gives you the list of all containers. After that you can start a command shell in the container by typing

docker exec -it <container_id> bash

this gives you access inside the container. But I don't think you need to enter the car type in any way, unless it's not listed in the car models database file?

**Comment by Hotzi57** [comment]
Thanks for the tip, I'll try it.
I also believe that my car type is already there, but you can also customize the car controller.

**Comment by koenigch** [comment]
Hi, i have an similar problem.
I need to change the price for the KWH, this is set in the config.ini file.
With docker exec -it <ADON NAME> /bin/bash i can see the files
But no editor like vi or nano or other editors will work. -> command not found
How can i  change values in the config.ini.

Thanks a lot
greats sindbard

**Reply by mitasa** [reply]
Edit the file outside the container and copy the edited file back to the container using: docker cp config.ini <container id>:/data/config.ini

**Comment by koenigch** [comment]
Hi mitasa,
thank you for your answer,
i´m not an expert in docker, (im a noob ;-) )
can you type the syntax for me in detail.
How can i copy the original ini out of the docker box?

thanks a lot
sindbard

**Reply by mitasa** [reply]
Looks like something was lost in my last post as I included the container_id in angle brackets to make myself clear. In the following, replace the container_id text with the actual container id.

First, find out the container id by typing "docker ps" and look for the line containing the PSA car controller.

Next, copy the config file out of the container by typing "docker cp container_id:/data/config.ini ."

Now you have the file in your working directory. Either edit it there using for example nano, or download it to your PC and edit it there & upload it back afterwards.

When the file has been edited, copy it back to the container using "docker cp config.ini container_id:/data/config.ini"

**Reply by MEvansDCC** [reply]
I don't seem to be able to do this in the HASSIO container, Would I be better off altering the INI and reinstalling somehow?
